### PR TITLE
refactor(testnode): create genTx on `Export`

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3662

Before, genTx transactions were created prematurely via [`WithValidators`](https://github.com/rootulp/celestia-app/blob/34cd4b34d99e27c5e8611abab6aaecc550499045/test/util/testnode/config.go#L123) which was problematic if a user tried to later override the chainID via `WithChainID` because the genTx transactions would use the default chain ID but the state machine would be created with a custom chain ID.

This PR refactors the genTx creation to happen only when `genesis.Export` is invoked which is _after_ a user configures a custom chain ID.

### CODE REVIEW

File: app/test/testnode_test.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+		genesis := genesis.NewDefaultGenesis().
+			WithChainID(chainID).
+			WithValidators(genesis.NewDefaultValidator(testnode.DefaultValidatorAccountName)).
+			WithConsensusParams(testnode.DefaultConsensusParams())
+
+		config := testnode.DefaultConfig()
+		config.WithChainID(chainID)
+		config.WithGenesis(genesis)
+		testnode.NewNetwork(t, config)
+	})
```
[the remark this answers is not in the document]
Comment (staheri14) [reply app/test/testnode_test.go:29]: What do you think about applying this suggestion of coderabbitai?

**Review by staheri14 — APPROVED** [review]

File: app/test/testnode_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (staheri14) [reply app/test/testnode_test.go:29]: Never mind, I think this is already handled internally by `NewNetwork`.

**Review by evan-forbes — APPROVED** [review]

File: test/util/testnode/config.go [hunk 2]
```diff
@@ -128,8 +128,8 @@ func DefaultConfig() *Config {
 		WithAppOptions(DefaultAppOptions()).
 		WithAppCreator(cmd.NewAppServer).
 		WithSuppressLogs(true).
-		WithConsensusParams(DefaultConsensusParams()).
-		WithSuppressLogs(true)
+		// TODO: consider removing this line because default consensus params is invoked above.
+		WithConsensusParams(DefaultConsensusParams())
```
Comment (cmwaters) [inline test/util/testnode/config.go:132]: I think we should remove this

**Review by cmwaters — APPROVED** [review]

File: test/util/testnode/config.go (hunk 2 above)
Comment (rootulp) [reply test/util/testnode/config.go:132]: https://github.com/celestiaorg/celestia-app/issues/3684

**Pull request merged**